### PR TITLE
[Bugfix] Make prompt_logprobs respect logprobs_mode setting

### PR DESCRIPTION
**Pull request by wojciech-wais**
prompt_logprobs always returned log_softmax of raw logits regardless of the logprobs_mode config, while output logprobs correctly respected it. This threads logprobs_mode through PromptLogprobsWorker (V2 model runner) and checks it in _get_prompt_logprobs_dict (V1 model runner) so prompt_logprobs honors the same mode as output logprobs.

Fixes #35832

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/v1/worker/gpu/sample/prompt_logprob.py [hunk 1]
```diff
@@ -195,13 +199,20 @@ def compute_prompt_logprobs_with_chunking(
...
         prompt_logits = logits_fn(prompt_hidden_states[start_idx:end_idx])
-        prompt_logprobs = compute_topk_logprobs(
+
+        # NOTE: For prompt logprobs, there is no "processing" step
+        # (no penalties, temperature, etc.), so "processed_logprobs" and
+        # "raw_logprobs" are equivalent. Both modes use log_softmax via
+        # compute_topk_logprobs. This matches the sampler behavior where
+        # the logprobs_mode selects between raw and processed logits
+        # before computing log_softmax.
+        prompt_logprobs_result = compute_topk_logprobs(
```
Comment (njhill) [inline vllm/v1/worker/gpu/sample/prompt_logprob.py]: what's the reason for renaming the variable?

File: vllm/v1/worker/gpu/model_runner.py [hunk 2]
```diff
@@ -196,7 +196,10 @@ def __init__(
             logprobs_mode=self.model_config.logprobs_mode,
             num_speculative_tokens=self.num_speculative_steps + 1,
         )
-        self.prompt_logprobs_worker = PromptLogprobsWorker(self.max_num_reqs)
+        self.prompt_logprobs_worker = PromptLogprobsWorker(
+            self.max_num_reqs,
+            logprobs_mode=self.model_config.logprobs_mode,
```
Comment (njhill) [inline vllm/v1/worker/gpu/model_runner.py]:
```suggestion
            self.max_num_reqs, logprobs_mode=self.model_config.logprobs_mode
```

File: vllm/v1/worker/gpu/sample/prompt_logprob.py [hunk 3]
```diff
@@ -184,6 +187,7 @@ def compute_prompt_logprobs_with_chunking(
     prompt_token_ids: torch.Tensor,
     prompt_hidden_states: torch.Tensor,
     logits_fn: Callable[[torch.Tensor], torch.Tensor],
+    logprobs_mode: LogprobsMode = "raw_logprobs",
```
Comment (njhill) [inline vllm/v1/worker/gpu/sample/prompt_logprob.py:212]: It doesn't look like this is used?

**Review by njhill — COMMENTED** [review]
Thanks @wojciech-wais!

File: vllm/v1/worker/gpu/sample/prompt_logprob.py (hunk 1 above)
Comment (wojciech-wais) [reply vllm/v1/worker/gpu/sample/prompt_logprob.py]: When applying the fix for the first issue there is no reason of renaming it. I used it for debugging.
This is fixed in the newer version of the code.

File: vllm/v1/worker/gpu/sample/prompt_logprob.py (hunk 3 above)
Comment (wojciech-wais) [reply vllm/v1/worker/gpu/sample/prompt_logprob.py:212]: Good catch — it was indeed threaded through but never acted on. Fixed: replaced the compute_topk_logprobs call (which always applies log_softmax via its triton kernel) with inline logic that conditionally applies F.log_softmax only for *_logprobs modes, and returns raw float32 logits for *_logits modes.

File: vllm/v1/worker/gpu/model_runner.py (hunk 2 above)
Comment (wojciech-wais) [reply vllm/v1/worker/gpu/model_runner.py]: Applied, thanks.

**Comment by wojciech-wais** [comment]
This PR needs significant conflict resolution due to 52 days since last update (conflicts in vllm/v1/worker/gpu/model_runner.py and vllm/v1/worker/gpu/sample/prompt_logprob.py). The manual testing I posted on Mar 4 shows all modes work correctly. Will need dedicated time to resolve the rebase conflicts, but the core functionality is sound. Marking for follow-up.

**Comment by wojciech-wais** [comment]
I've successfully resolved merge conflicts and rebased on latest main. The fix now properly integrates with the current codebase architecture.

The manual testing I posted on Mar 4 shows all modes work correctly (`raw_logprobs`, `raw_logits`, `processed_logprobs`, `processed_logits`).

It's ready for review again.

**Comment by aoshen02** [comment]
Hi, sorry for the late response and in case you are not working on this pr, I've create another one and list you the author. https://github.com/vllm-project/vllm/pull/47680

**Comment by aoshen02** [comment]
Hi Wojciech, we plan to merge 47680 and list you the first author, what do you think? If it is ok for you, would you like to do some code review for me.

**Comment by wojciech-wais** [comment]
@aoshen02 
I've synchronized the code with the latest main, but I also saw your PR.
Your on it and your PR is more up-to-date with the latest main so let go with your approach.

**Comment by wojciech-wais** [comment]
> Hi Wojciech, we plan to merge 47680 and list you the first author, what do you think? If it is ok for you, would you like to do some code review for me.

Fine for me.

I am looking into your PR now.

**Comment by aoshen02** [comment]
> > Hi Wojciech, we plan to merge 47680 and list you the first author, what do you think? If it is ok for you, would you like to do some code review for me.
> 
> Fine for me.
> 
> I am looking into your PR now.

Cool, thanks for the review.

**Comment by njhill** [comment]
Superseded by https://github.com/vllm-project/vllm/pull/47680.
